gtsfm/evaluation/metrics_report.py

```diff
@@ -110,6 +146,110 @@ def create_plots_for_distributions(metrics_dict: Dict[str, Any]) -> str:
     return fig.to_html(full_html=False, include_plotlyjs="cdn")
 
 
+def create_plots_for_distributions_and_compare(
+    metrics_dict: Dict[str, Any], other_pipeline_metrics_dicts: List[Dict], pipeline_names: List[str]
+) -> str:
+    """Creates plots for 1D distribution metrics using plotly, and converts them to HTML.
+
+    The plots are arranged in a grid, with each row having SUBPLOTS_PER_ROW (3) columns.
+    For a certain metric, these can be either histogram or box according to the metric's property.
+
+    Args:
+        metrics_dict: A dict, where keys are names of metrics and values are
+        the dictionary representation of the metric.
+        other_pipeline_metrics_dicts: metrics_dicts for other SfM pipelines
+        pipeline_names: A list of other SfM pipeline names
+    Returns:
+        Plots in a grid converted to HTML as a string.
+    """
+    distribution_metrics = []
+    # Separate all the 1D distribution metrics.
+    for metric, value in metrics_dict.items():
+        if isinstance(value, dict):
+            all_nan_summary = all(np.isnan(v) for v in value[metrics.SUMMARY_KEY].values())
+            if not all_nan_summary:
+                distribution_metrics.append(metric)
+    if len(distribution_metrics) == 0:
+        return ""
+    fig_html = ""
+    # Setup figure layout - number of rows and columns.
+    num_rows = 1
+    subplots_per_row = 1
+    # Iterate over all metrics.
+    for metric_name, metric_value in metrics_dict.items():
+        fig = psubplot.make_subplots(
+            rows=num_rows, cols=subplots_per_row, subplot_titles=(metric_name,)
+        )  # without trailing comma in subplot_titles, title is lost.
+        fig.update_layout({"height": 512 * num_rows, "width": 1024, "showlegend": True})
+        # j = 0
+        # Check if this is a 1D distribution metric and has a summary.
+        if metric_name not in distribution_metrics or metrics.SUMMARY_KEY not in metric_value:
+            continue
+        row = 1
+        col = 1
+        # Histogram metrics are plotted directly from summary.
+        if "histogram" in metric_value[metrics.SUMMARY_KEY]:
+            histogram = metric_value[metrics.SUMMARY_KEY]["histogram"]
+            fig.add_trace(
```

Comment:
is there some way we can reduce the amount of code in this function by having another function to create the plots, which gets called with both gtsfm metrics and other pipeline metrics (i.e the plotting code is the same and does not need repetition).

Reply:
Added a function called make_fig() to reuse code.